### PR TITLE
Search m

### PR DESCRIPTION
**Pull request by ekansa**
### brief description of changes
I added some documentation about search features, including use of the "negation" (exclude) feature, some special text pattern matching, and some topics in Advanced search. We didn't have anything about this topic until now, so I thought it would be a good addition.

#### issues addressed
https://github.com/archesproject/arches-docs/issues/225

#### further comments
Search is a big topic. We may want to document more aspects of search incrementally with multiple pull  requests. This initial PR should document the essentials first, then we can add more nuance and details later.

You can preview the search documentation added in this PR here:
https://arches.readthedocs.io/en/search_m/user-guide/searching/

Thanks @apeters for the pointers about the search operators!

---

This box **must** be checked
- [x] the PR branch was originally made from the base branch

This box **should** be checked
- [x] after these changes the docs build locally without error

This box **should only** be checked you intend to follow through on it (we can do it on our end too)
- [ ] I will `cherry-pick` all commits in this PR into other branches that should have them _after_ this PR is merged

### CODE REVIEW

**Comment by mradamcox** [comment]
@ekansa Glad to see this, it looks really good overall, as you say, a good start. A couple notes though:

Ticket #222 that you referenced is actually geared toward documenting how to make custom search filters _a la_ "extensions", not how to use the default search filters that come with Arches. The ticket you probably meant is #225. Looking at the latter, perhaps you it wouldn't be to difficult too much more work to address #185 in this PR as well?

Also, just wanted to make sure you're aware of the in-app help available in the Arches Search interface (`?` button in top-right). There is a good bit of information in there on the different default search tools that are available, but it hasn't been updated for a while, so worth a lookover before copying into these docs. That template is here: https://github.com/archesproject/arches/blob/master/arches/app/templates/help/search-help.htm, and, this is a good reminder that there is a fair bit of information in those help templates that may not actually be in the User Guide. Worth looking through for sure.

**Review by mradamcox — COMMENTED** [review]
See comment on main thread (no specific changes requested, I'd just request a quick lookover the idea of closing 185 out within this PR as well).

File: docs/user-guide/searching.txt [hunk 1]
```diff
@@ -2,4 +2,32 @@
...
+
+
+    **Equals**: This is a complete phrase search. It will only match full words in the exact order given. For example, if the search term is "stock book” it will find the value “Knoedler Stock Book 4”. It will *NOT* find “... books in stock ...“, or “... stocking new books ...”
+
+    **Wildcard**: This assumes the user is going to supply a search phrase exactly as they want. For example, a search for "st?ck" will find the values of “stock”, or “Stick”.
+    It will *NOT* find “The car is stuck in the woods”, because the “stuck” is surrounded by other words. If you need a "contains-like" query, then you need to supply leading and trailing asterisks. So, "\*st?ck\*" *WILL* find “The car is stuck in the woods”.
+
+In all the examples above we quote the search string for clarity. In the Arches application you wouldn’t need to put your search terms inside quotes.
```
Comment (apeters) [inline docs/user-guide/searching.txt:24]: I think we should make mention of the use of quotes as a way to define an "exact" search.
An exact search only finds values that match then entire search string exactly, including case.

**Review by apeters — CHANGES_REQUESTED** [review]

File: docs/user-guide/searching.txt (hunk 1 above)
Comment (ekansa) [reply docs/user-guide/searching.txt:33]: @mradamcox Great! I added a short section and animation about the in-app search guide too.

**Pull request merged**